bitwarden_license/test/Scim.IntegrationTest/Controllers/v2/GroupsControllerTests.cs

```diff
@@ -0,0 +1,464 @@
+                Members = null,
+                Schemas = null
+            };
+
+            var context = await _factory.GroupsPostAsync(organizationId, model);
+
+            Assert.Equal(StatusCodes.Status201Created, context.Response.StatusCode);
+
+            var databaseContext = _factory.GetDatabaseContext();
+            Assert.Equal(4, databaseContext.Groups.Count());
+            Assert.True(databaseContext.Groups.Any(g => g.Name == "New Group"));
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public async Task Post_InvalidDisplayName_BadRequest(string displayName)
+        {
+            var organizationId = ScimApplicationFactory.TestOrganizationId1;
+            var model = new ScimGroupRequestModel
+            {
+                DisplayName = displayName,
+                ExternalId = null,
+                Members = null,
+                Schemas = null
+            };
+
+            var context = await _factory.GroupsPostAsync(organizationId, model);
+
+            Assert.Equal(StatusCodes.Status400BadRequest, context.Response.StatusCode);
+        }
+
+        [Fact]
+        public async Task Post_ExistingExternalId_Conflict()
+        {
+            var organizationId = ScimApplicationFactory.TestOrganizationId1;
+            var model = new ScimGroupRequestModel
+            {
+                DisplayName = "New Group",
+                ExternalId = "A",
+                Members = null,
+                Schemas = null
+            };
+
+            var context = await _factory.GroupsPostAsync(organizationId, model);
+
+            Assert.Equal(StatusCodes.Status409Conflict, context.Response.StatusCode);
+
+            var databaseContext = _factory.GetDatabaseContext();
+            Assert.Equal(3, databaseContext.Groups.Count());
+            Assert.False(databaseContext.Groups.Any(g => g.Name == "New Group"));
+        }
+
+        [Fact]
+        public async Task Put_ChangeName_Success()
+        {
+            var newGroupName = "Test Group 1 New Name";
+            var organizationId = ScimApplicationFactory.TestOrganizationId1;
+            var id = ScimApplicationFactory.TestGroupId1;
+            var model = new ScimGroupRequestModel
+            {
+                DisplayName = newGroupName,
+                ExternalId = "AA",
+                Members = new List<ScimGroupRequestModel.GroupMembersModel>(),
+                Schemas = new List<string>()
+            };
+
+            var context = await _factory.GroupsPutAsync(organizationId, id, model);
+
+            Assert.Equal(StatusCodes.Status200OK, context.Response.StatusCode);
+
+            var databaseContext = _factory.GetDatabaseContext();
+            Assert.Equal(3, databaseContext.Groups.Count());
+
+            var firstGroup = databaseContext.Groups.FirstOrDefault(g => g.Id.ToString() == id);
+            Assert.Equal(newGroupName, firstGroup.Name);
```

Comment:
What is the expected behaviour re: members? I haven't looked into it much, but given it's a PUT and not a PATCH, I assume members should be overwritten completely, right?